### PR TITLE
Example UID entrypoint to enable restricted SCC

### PR DESCRIPTION
**Pull request by shawn-hurley**
> When running in OpenShift under the standard restricted SCC the uid is random, so it's necessary to grant access to the anyuid SCC for the current operator to work.

> This PR adds a custom entrypoint to the example operator image that patches /etc/passwd at runtime with the container uid so that ansible (ssh) works with the default (restricted) SCC. This approach is used by other images (e.g. openshift-ansible).

> This type support should probably eventually be built in the base image so that this customization is not needed. Right now ansible-runner seems to have some partial support for this but it does not work yet.

@codificat

### CODE REVIEW

File: bin/entrypoint [hunk 1]
```diff
@@ -0,0 +1,13 @@
+#!/bin/bash -e
+
+# Patch /etc/passwd file with the current user info.
+# The current user's entry must be correctly defined in this file in order for
+# the `ssh` command to work within the created container.
```
Comment (mhrivnak) [inline bin/entrypoint]: There are perfectly legit reasons to do this, but ssh inside the container isn't normally one of them (it's usually frowned upon). I'd leave that part out so as not to raise anyone else's eyebrows, and instead maybe just reference the openshift docs that explain it.

https://docs.openshift.com/container-platform/3.10/creating_images/guidelines.html#openshift-specific-guidelines

File: tmp/build/Dockerfile [hunk 2]
```diff
@@ -8,5 +8,17 @@ RUN echo "localhost ansible_connection=local" > /etc/ansible/hosts \
     && echo 'roles_path = /opt/ansible/roles' >> /etc/ansible/ansible.cfg \
     && echo 'library = /usr/share/ansible/openshift' >> /etc/ansible/ansible.cfg
 
+ENV OPERATOR=/usr/local/bin/ansible-operator \
+    USER_UID=1001 \
+    USER_NAME=runner \
```
Comment (mhrivnak) [inline tmp/build/Dockerfile]: Is `runner` a better username than `ansible-operator`? I'm not opposed necessarily, but just interested in the reasoning since that change doesn't seem related to the rest of the PR.

**Comment by shawn-hurley** [comment]
Thanks again @codificat
